sklearn/calibration.py

```diff
@@ -753,3 +755,209 @@ def calibration_curve(y_true, y_prob, *, normalize=False, n_bins=5,
+        self.y_prob = y_prob
+        self.estimator_name = estimator_name
+
+    def plot(self, ax=None, *, name=None, ref_line=True, **kwargs):
+        """Plot visualization.
+
+        Extra keyword arguments will be passed to matplotlib's `plot`.
+
+        Parameters
+        ----------
+        ax : Matplotlib Axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        name : str, default=None
+            Name of calibration curve for labeling. If `None`, use the
+            name of the estimator.
+
+        ref_line : bool, default=True
+            If `True`, plots a reference line representing a perfectly
+            calibrated classifier.
+
+        **kwargs : dict
+            Keyword arguments to be passed to :func:`matplotlib.pyplot.plot`.
+
+        Returns
+        -------
+        display : :class:`~sklearn.calibration.CalibrationDisplay`
+            Object that stores computed values.
+        """
+        check_matplotlib_support("CalibrationDisplay.plot")
+        import matplotlib.pyplot as plt
+
+        if ax is None:
+            fig, ax = plt.subplots()
+
+        name = self.estimator_name if name is None else name
+
+        line_kwargs = {}
+        if name is not None:
+            line_kwargs["label"] = name
+        line_kwargs.update(**kwargs)
+
+        existing_ref_line = ('Perfectly calibrated' in
+                             ax.get_legend_handles_labels()[1])
+        if ref_line and not existing_ref_line:
+            ax.plot([0, 1], [0, 1], "k:", label="Perfectly calibrated")
+        self.line_ = ax.plot(self.prob_pred, self.prob_true, "s-",
+                             **line_kwargs)[0]
+
+        if "label" in line_kwargs:
+            ax.legend(loc="lower right")
+
+        ax.set(xlabel="Mean predicted probability",
+               ylabel="Fraction of positives")
+
+        self.ax_ = ax
+        self.figure_ = ax.figure
+        return self
+
+
+def plot_calibration_curve(estimator, X, y, *,
+                           n_bins=5, strategy='uniform',
+                           name=None, ref_line=True, ax=None, **kwargs):
+    """Plot calibration curve for binary classifiers.
+
+    The average predicted probability for each bin is plotted on the x-axis
+    and the fraction of positive classes in each bin is plotted on the y-axis.
+
+    Extra keyword arguments will be passed to :func:`matplotlib.pyplot.plot`.
+
+    Read more in the :ref:`User Guide <calibration>`.
+
+    Parameters
+    ----------
+    estimator : estimator instance
+        Fitted classifier or a fitted :class:`~sklearn.pipeline.Pipeline` in
+        which the last estimator is a classifier. The classifier must
+        have a :term:`predict_proba` method; set `probability=True` for
+        :mod:`~sklearn.svm` estimators (see:
+        :ref:`User Guide <scores_probabilities>`) or use
+        :class:`~sklearn.calibration.CalibratedClassifierCV`.
```

Comment:
Good point. Are there any/many classifiers that do not have a `predict_proba` method (or option to add it) ? I only know of `LinearSVC`. If there are a few, it might be worth mentioning what to do when the classifer doesn't have a `predict_proba`, in `calibration.rst` ?

Reply:
perceptron and some SGDCLassifier depending on the loss.